src/pineko/scale_variations.py

```diff
@@ -0,0 +1,154 @@
+"""Module to generate scale variations."""
+import pineappl
+import rich
+from eko import beta
+
+from . import check
+
+
+def ren_sv_contrib(m, delta, logpart, which_part, nf):
+    """Return the ren_sv contribution relative to the requested log power and perturbative order contribution (which_part).
+
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    delta : int
+        difference between asked order and m
+    logpart : int
+        power of the log asked
+    which_part : int
+        asked perturbative order contribution
+    nf : int
+        number of active flavors
+
+    Returns
+    -------
+    float
+        renormalization scale variation contribution
+    """
+    if delta == 0:
+        return 0.0
+    elif delta == 1:
+        return -m * beta.beta_qcd((2, 0), nf)
+    elif delta == 2:
+        if which_part == 0:
+            if logpart == 1:
+                return -m * beta.beta_qcd((3, 0), nf)
+            else:
+                return 0.5 * m * (m + 1) * (beta.beta_qcd((2, 0), nf) ** 2)
+        else:
+            return -(m + 1) * beta.beta_qcd((2, 0), nf)
+
+
+def compute_scalefactor(m, nec_order, to_construct_order, nf, kR):
+    """Compute the factor of renormalization scale variation.
+
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    nec_order : tuple(int)
+        tuple of the order for which the sv contribution is asked for
+    to_contruct_order : tuple(int)
+        tuple of the sv order to be constructed
+    nf : int
+        number of active flavors
+    kR : float
+        log of the ratio between renormalization scale and Q
+
+    Returns
+    -------
+    float
+        full contribution of ren sv
+    """
+    delta = to_construct_order[0] - m
+    logpart = to_construct_order[2]
+    return (kR**logpart) * ren_sv_contrib(m, delta, logpart, nec_order[0] - m, nf)
+
+
+def compute_map_of_necessary_orders(m, delta):
+    """Compute a dictionary with all the necessary order to compute in order to have the full renormalization scale variation.
+
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    delta : int
+        difference between asked order and m
+
+    Returns
+    -------
+    dict(tuple(int))
+        description of all the needed orders
+    """
+    nec_orders = {}
+    for delt in range(delta):
+        nec_orders[(m + delta, 0, delt + 1, 0)] = [
+            (m + de, 0, 0, 0) for de in range(delta - delt)
+        ]
+    return nec_orders
+
+
+def create_svonly_grid(grid, order, new_order, scalefactor):
+    """Create a grid containing only the renormalization scale variations at a given order for a grid."""
+    # Retrieve parameters to create new grid
+    bin_limits = [
+        float(bin) for bin in range(grid.raw.bins() + 1)
+    ]  # +1 because I don't know
```

Comment:
This comment is not reassuring...
(maybe leave at least a `TODO` there)

Reply:
The `+1` explanation is that `n` bins have `n+1` bin limits, and `range` generates numbers from a half-open interval (`range(n)` generates `n` numbers).

Reply:
Ok, this makes perfect sense, and it can be a better comment :)